UnityProject/Assets/Scripts/Core/Input System/InteractionV2/TileInteraction/TileApply.cs

```diff
@@ -38,17 +38,20 @@ public class TileApply : Interaction
 	/// </summary>
 	public GameObject HandObject => UsedObject;
 
-	private readonly Vector2 targetVector;
+	private readonly Vector2 targetPosition;
+
 
 	/// <summary>
 	/// Targeted world position deduced from target vector and performer position.
 	/// </summary>
-	public Vector2 WorldPositionTarget => (Vector2)Performer.transform.position + targetVector;
+	public Vector2 WorldPositionTarget => TargetPosition.To3().ToWorld(Performer.RegisterTile().Matrix);
 
 	/// <summary>
 	/// Vector pointing from the performer to the targeted position. Set to Vector2.zero if aiming at self.
 	/// </summary>
-	public Vector2 TargetVector => targetVector;
+	public Vector2 TargetPosition => targetPosition;
+
+	public Vector2 TargetVector => Performer.transform.localPosition.To2() - targetPosition;
```

Comment:
```suggestion
	/// <summary>Vector pointing from the performer's position to the target position.</summary>
	public Vector2 TargetVector => targetPosition - Performer.transform.localPosition.To2();
```

This one was backwards, right?